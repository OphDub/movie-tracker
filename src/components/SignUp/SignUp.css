.error {
  border: 1px solid red;
}

.login-error {
  font-size: .8em;
  color: red;
}