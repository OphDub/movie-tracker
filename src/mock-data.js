 export const cleanMovieArray = [
    {
      movie_id: 399055,
      title: 'The Shape of Water',
      vote_average: 7.4,
      vote_count: 794,
      overview: 'An other-worldly story, set against the backdrop of Cold War era America circa 1962, where a mute janitor working at a lab falls in love with an amphibious man being held captive there and devises a plan to help him escape.',
      release_date: '2017-12-08',
      poster_path: 'https://image.tmdb.org/t/p/w500//iLYLADGA5oKGM92Ns1j9CDgk3iI.jpg'
    },
    {
      "movie_id": 284054,
      "title": "Black Panther",
      "vote_average": 5,
      "vote_count": 72,
      "overview": "T'Challa, the new ruler of the advanced kingdom of Wakanda, must defend his land from being torn apart by enemies from outside and inside the country.",
      "release_date": "2018-02-13",
      "poster_path": "https://image.tmdb.org/t/p/w500//bLBUCtMQGJclH36clliPLmljMys.jpg"
    }
] 

export const dirtyMovieObject = { results: [
    { 
      vote_average: 7.4,
      overview: 'An other-worldly story, set against the backdrop of Cold War era America circa 1962, where a mute janitor working at a lab falls in love with an amphibious man being held captive there and devises a plan to help him escape.',
      vote_count: 794,
      title: 'The Shape of Water',
      poster_path: '/iLYLADGA5oKGM92Ns1j9CDgk3iI.jpg',
      release_date: '2017-12-08',
      id: 399055,
      randomnumber: 24341324142134,
    },
    {
      "title": "Black Panther",
      "vote_count": 72,
      "thing": "asdfasdfasdf",
      "id": 284054,
      "overview": "T'Challa, the new ruler of the advanced kingdom of Wakanda, must defend his land from being torn apart by enemies from outside and inside the country.",
      "release_date": "2018-02-13",
      "poster_path": "/bLBUCtMQGJclH36clliPLmljMys.jpg",
      "vote_average": 5,
    }
]}

export const oneMovie = {
  movie_id: 284054,
  title: 'Coco',
  vote_average: 7.7,
  vote_count: 1987,
  overview: 'Despite his familyâ€™s baffling generations-old ban on music, Miguel dreams of becoming an accomplished musician like his idol, Ernesto de la Cruz. Desperate to prove his talent, Miguel finds himself in the stunning and colorful Land of the Dead following a mysterious chain of events. Along the way, he meets charming trickster Hector, and together, they set off on an extraordinary journey to unlock the real story behind Miguel\'s family history.',
  posterImage: 'https://image.tmdb.org/t/p/w500//eKi8dIrr8voobbaGzDpe8w0PVbC.jpg'
}

export const userData = {
  id: 2,
  name: 'brophus park',
  email: 'bros@gmail.com',
  password: 'pass',
}

export const movieRawData = {
  data: {
      movie_id: 399055,
      title: 'The Shape of Water',
      vote_average: 7.4,
      vote_count: 794,
      overview: 'An other-worldly story, set against the backdrop of Cold War era America circa 1962, where a mute janitor working at a lab falls in love with an amphibious man being held captive there and devises a plan to help him escape.',
      release_date: '2017-12-08',
      poster_path: 'https://image.tmdb.org/t/p/w500//iLYLADGA5oKGM92Ns1j9CDgk3iI.jpg'
    }
}
