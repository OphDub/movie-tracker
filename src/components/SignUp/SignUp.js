import React, { Component } from 'react';
import { connect } from 'react-redux';
import { activeUserAction } from '../../actions/actionIndex';
import { validateUser } from '../../api.js';
import { signUpUser, getAllUsers } from '../../api.js';
import './SignUp.css';

export class SignUp extends Component {
  constructor() {
    super()
    this.state = {
      name: '',
      email: '',
      password:'',
      errorStatus: false
    }
  }

  handleChange = (e) => {
    const { name, value } = e.target;
    this.setState({[name]:value})
  }

  handleSignUp = async (e) => {
    e.preventDefault();
    const { name, email, password } = this.state;
    const allUsers = await getAllUsers();
    const userExists = await allUsers.find(user => user.email === email);

    if (userExists) {
      this.emailTaken(true);
    } else {
      await signUpUser(email, password, name)
      const validate = await validateUser(name, password);

      this.props.handleLogin(validate.data)
    }

    return userExists
  }

  emailTaken = (warning) => {
    if (warning) {
      this.setState({errorStatus: true})
    }
  }

  render () {
    return (
      <div>
        <form type="submit">
          <input
            value={this.state.name}
            onChange={this.handleChange}
            name="name"
            type="text"
            placeholder="Enter your name"
          />
          <input
            className={!this.state.errorStatus ? "" : "error"}
            value={this.state.email}
            onChange={this.handleChange}
            name="email"
            type="text"
            placeholder="Enter Email"
          />
          <input
            value={this.state.password}
            onChange={this.handleChange}
            name="password"
            type="text"
            placeholder="Enter your password"
          />
          <button
            type="submit"
            onClick={this.handleSignUp}>
            Sign Up
          </button>
        </form>
       
       { this.state.errorStatus && 
        <p className="login-error">Sorry, an account with this e-mail already exists</p>
       }
      </div>
    )
  }
}

const mapDispatch = (dispatch) => ({
  handleLogin: (user) => dispatch(activeUserAction(user)),
})

export default connect(null, mapDispatch)(SignUp)