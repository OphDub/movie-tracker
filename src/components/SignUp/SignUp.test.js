/* eslint-disable */
import React from 'react';
import { SignUp } from './SignUp';
import { shallow } from 'enzyme';
import { userData } from '../../mock-data.js';

describe('SignUp', () => {
  let wrapper;

  beforeEach( () => {
    const mockFn = jest.fn();

    wrapper = shallow(
      <SignUp handleLogin={mockFn}/>);
  });

  it('should match the snapshot test', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('handleChange should update state when it is called', () => {
    const mockEvent = { target: {name: 'name' ,value: 'brophus park'}};

    wrapper.instance().handleChange(mockEvent);
    wrapper.update();

    expect(wrapper.state().name).toEqual(mockEvent.target.value);
  });

  it('emailTaken should update state to match the value it is given', () => {
    const truthyValue = true;
    const falsyValue = false;

    wrapper.instance().emailTaken(truthyValue);
    wrapper.update();

    expect(wrapper.state().errorStatus).toEqual(true);

    wrapper.instance().emailTaken(falsyValue);
    wrapper.update();

    expect(wrapper.state().errorStatus).toEqual(false);
  });
});