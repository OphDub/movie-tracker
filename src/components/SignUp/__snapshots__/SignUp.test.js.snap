// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp should match the snapshot test 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SignUp />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3
          className="signup-prompt"
        >
          New to MovieTracker? Sign-up for an account here:
        </h3>,
        <form
          className="signup-form"
          type="submit"
        >
          <input
            name="name"
            onChange={[Function]}
            placeholder="Enter your name"
            type="text"
            value=""
          />
          <input
            className=""
            name="email"
            onChange={[Function]}
            placeholder="Enter Email"
            type="text"
            value=""
          />
          <input
            name="password"
            onChange={[Function]}
            placeholder="Enter your password"
            type="password"
            value=""
          />
          <button
            className="signup-btn"
            onClick={[Function]}
            type="submit"
          >
            Sign Up
          </button>
        </form>,
        false,
      ],
      "className": "signup",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "New to MovieTracker? Sign-up for an account here:",
          "className": "signup-prompt",
        },
        "ref": null,
        "rendered": "New to MovieTracker? Sign-up for an account here:",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              name="name"
              onChange={[Function]}
              placeholder="Enter your name"
              type="text"
              value=""
            />,
            <input
              className=""
              name="email"
              onChange={[Function]}
              placeholder="Enter Email"
              type="text"
              value=""
            />,
            <input
              name="password"
              onChange={[Function]}
              placeholder="Enter your password"
              type="password"
              value=""
            />,
            <button
              className="signup-btn"
              onClick={[Function]}
              type="submit"
            >
              Sign Up
            </button>,
          ],
          "className": "signup-form",
          "type": "submit",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "name",
              "onChange": [Function],
              "placeholder": "Enter your name",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "",
              "name": "email",
              "onChange": [Function],
              "placeholder": "Enter Email",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "password",
              "onChange": [Function],
              "placeholder": "Enter your password",
              "type": "password",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Sign Up",
              "className": "signup-btn",
              "onClick": [Function],
              "type": "submit",
            },
            "ref": null,
            "rendered": "Sign Up",
            "type": "button",
          },
        ],
        "type": "form",
      },
      false,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3
            className="signup-prompt"
          >
            New to MovieTracker? Sign-up for an account here:
          </h3>,
          <form
            className="signup-form"
            type="submit"
          >
            <input
              name="name"
              onChange={[Function]}
              placeholder="Enter your name"
              type="text"
              value=""
            />
            <input
              className=""
              name="email"
              onChange={[Function]}
              placeholder="Enter Email"
              type="text"
              value=""
            />
            <input
              name="password"
              onChange={[Function]}
              placeholder="Enter your password"
              type="password"
              value=""
            />
            <button
              className="signup-btn"
              onClick={[Function]}
              type="submit"
            >
              Sign Up
            </button>
          </form>,
          false,
        ],
        "className": "signup",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "New to MovieTracker? Sign-up for an account here:",
            "className": "signup-prompt",
          },
          "ref": null,
          "rendered": "New to MovieTracker? Sign-up for an account here:",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                name="name"
                onChange={[Function]}
                placeholder="Enter your name"
                type="text"
                value=""
              />,
              <input
                className=""
                name="email"
                onChange={[Function]}
                placeholder="Enter Email"
                type="text"
                value=""
              />,
              <input
                name="password"
                onChange={[Function]}
                placeholder="Enter your password"
                type="password"
                value=""
              />,
              <button
                className="signup-btn"
                onClick={[Function]}
                type="submit"
              >
                Sign Up
              </button>,
            ],
            "className": "signup-form",
            "type": "submit",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "name",
                "onChange": [Function],
                "placeholder": "Enter your name",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "",
                "name": "email",
                "onChange": [Function],
                "placeholder": "Enter Email",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "password",
                "onChange": [Function],
                "placeholder": "Enter your password",
                "type": "password",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Sign Up",
                "className": "signup-btn",
                "onClick": [Function],
                "type": "submit",
              },
              "ref": null,
              "rendered": "Sign Up",
              "type": "button",
            },
          ],
          "type": "form",
        },
        false,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
